packages/components/modules/messages/MessagesList/MessagesGroup/MessageItem/index.tsx

```diff
@@ -13,17 +16,80 @@ const MessageItem: FC<MessageItemProps> = ({ messageRef, isFirstGroupedMessage }
   const { currentProfile } = useCurrentProfile()
   const message = useFragment(MessageItemFragment, messageRef)
   const isOwnMessage = currentProfile?.id === message?.profile?.id
+  const messageCardRef = useRef<HTMLDivElement>(null)
 
-  return (
-    <MessageItemContainer isOwnMessage={isOwnMessage} isFirstGroupedMessage={isFirstGroupedMessage}>
+  const [isEditMode, setIsEditMode] = useState(false)
+
+  const renderMessageContent = () => {
+    if (isEditMode) {
+      return <MessageUpdate message={message} onCancel={() => setIsEditMode(false)} />
+    }
+
+    return (
       <Typography
         variant="body2"
         color={isOwnMessage ? 'text.primary' : 'primary.contrastText'}
-        sx={{ maxWidth: '100%', wordWrap: 'break-word' }}
+        sx={{
+          maxWidth: '100%',
+          whiteSpace: 'pre-wrap',
+          wordBreak: 'normal',
+          overflowWrap: 'anywhere',
+        }}
       >
         {message?.content}
       </Typography>
-    </MessageItemContainer>
+    )
+  }
+
+  return (
+    <ActionsOverlay
+      title="message"
+      actions={[
+        {
+          disabled: false,
+          icon: <CopyIcon />,
+          label: 'Copy',
+          onClick: () => {}, // TODO: Implement copy message
+          hasPermission: true,
+        },
+        {
+          disabled: false,
+          icon: <PenEditIcon />,
+          label: 'Edit',
+          onClick: () => {
+            setIsEditMode(true)
+          },
+          hasPermission: isOwnMessage,
+          closeOnClick: true,
+        },
+        {
+          disabled: false,
+          icon: <DownloadIcon />,
+          label: 'Download Attachments',
+          onClick: () => {}, // TODO: Implement download attachments
+          hasPermission: true,
+        },
+      ]}
+      useTreedotsMenuOverlay
+      enableDelete
+      handleDeleteItem={() => {}} // TODO: Implement delete message
+      isDeletingItem={false}
+      ContainerProps={{
+        width: '100%',
+        display: 'flex',
+        flexDirection: isOwnMessage ? 'row' : 'row-reverse',
+        justifyContent: 'flex-end',
+      }}
```

Comment:
if we apply the suggestion of creating a `ThreeDotsMenuHoverOverlay` then all customization logic would be there, we would only need to pass `hoverOverlayMode === "three-dots-menu"`

Reply:
I think we still need at least the flexDirection.

because this style is specific to the messages and changes according to the isOwnMessage variable